Readability.js

```diff
@@ -1739,30 +1803,45 @@ Readability.prototype = {
   /* convert images and figures that have properties like data-src into images that can be loaded without JS */
   _fixLazyImages: function (root) {
     this._forEachNode(this._getAllNodesWithTag(root, ["img", "picture", "figure"]), function (elem) {
+      // In some sites (e.g. Kotaku), they put 1px square image as data uri in the src attribute.
+      // So, here we check if the data uri is too short, just might as well remove it.
+      if (elem.src && elem.src.startsWith("data:")) {
+        // I don't have any source but I guess if image is less than 100 bytes it will be too
+        // small, therefore it might be placeholder image. With that said, I will use 100B
+        // as threshold (or 133B after encoded to base64).
+        var b64starts = elem.src.indexOf("base64,") + 7;
```

Comment:
... though I guess that still treats base64 svgs as content-less. We can try and extract the mimetype and make an exception for it, but not sure that's worth it...

Reply:
I've added it at [23c670](https://github.com/mozilla/readability/pull/590/commits/23c670701e1dfafc002b6be926b5f93478109bb1). However, I made a little change from your regex into this :

```
^data:\s*(\S+)\s*;\s*base64\s*,
```

This is done because I found out several things about data URL :

- Media type according to [IANA](https://www.iana.org/assignments/media-types/media-types.xhtml) shouldn't have any space, so `(\S+)` is enough to capture it.
- Apparently media type in data URL could be surrounded by spaces.
- Same with `base64` could be surrounded by spaces as well, and data URL will still works.

Since media type in my regex is in capture group, we can use it to check for base64-encoded SVG file.

Reply:
I also added additional check in [be95939](https://github.com/mozilla/readability/pull/590/commits/be95939183235b2bfbf4fcd9cec5ebc7a3a00ede) to make sure that an image has other attributes beside `src` which contains image. If `src` is the only one that contains image, then it's important no matter how small it is.

Reply:
Can we make sure the capturing group explicitly does not include the `;` or `,`, so it doesn't have to backtrack potentially a lot (if the data URI is long) ? Ie `([^\s;,]+)` for the capturing expression would work, I believe.

Reply:
Sorry it took so long. I've added it in [2280afc](https://github.com/mozilla/readability/pull/590/commits/2280afc581cd98fccebf33f4c2c6a418b70ce118)

Reply:
No problem, thank you very much for all your work on this!